.github/workflows/build_and_test.yml

```diff
@@ -72,14 +81,18 @@ jobs:
           java-version: 21
           distribution: 'adopt'
           cache: 'gradle'
-      - name: start minikube
+      - name: Set Up Minikube
+        uses: medyagh/setup-minikube@v0.0.18
+        with:
+          cache: true
+          kubernetes-version: ${{ matrix.kube-version }}
+          cpu: 4
```

Comment:
It seems that Apache Spark community decided to use `2` based on the historic stability. This new value, `4`, looks larger than before. Is this okay?

https://github.com/apache/spark/blob/33059396bd2d529d499498106f51a42328b457ca/.github/workflows/build_and_test.yml#L1121-L1122

```
          # Github Action limit cpu:2, memory: 6947MB, limit to 2U6G for better resource statistic
          minikube start --cpus 2 --memory 6144
```

Reply:
Changed to cpu 2.

I was thinking once we add more e2e test, the Minikube might not large enough to run these tests in parallel.